### PR TITLE
Improve claim-backend logging

### LINKED ISSUE #62: Improve backend logs
### Expected behavior
token claiming backed should have good API logs with
* Request and Response Data
* Timestamp
* Client Information
* Performance Metrics
* Error Logging

### Actual behavior
The log is not organized in JSON and does not have sufficient information

### PR DESCRIPTION
**Pull request by Phanco**
### What was the problem?

This PR resolves #62

### How was it solved?
4 middlewares added: 2 Express Middleware and 2 RPC Middleware
1. `expressLogger`
2. `expressErrorHandler`
3. `rpcLogger`
4. `rpcErrorHandler`

Both `*ErrorHandler` are used to suppress error made by client. `expressLogger` logs data at HTTP level, info such as IP, user-agent, response time are recorded. `rpcLogger` logs with request and response from RPC API.

Due to the design of express, it requires some over-engineering to get the response at HTTP level, and on the RPC side there is also no HTTP data. That's why 2 middlewares are used.

### How was it tested?
- Testing backend with every logging level
- Testing each RPC endpoints
- Testing with valid and invalid data
Logging works as expected

### CODE REVIEW

File: packages/claim-backend/README.md [hunk 1]
```diff
@@ -11,20 +11,21 @@ Apart from Node version 18, Claim Backend also requires:
...
+| param                | Description                                                                                                                                          | Required                  |
+| -------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------- |
+| BACKEND_HOST         | IP used by backend (Default: 127.0.0.1)                                                                                                              | false                     |
+| BACKEND_PORT         | Port used by backend (Default: 3000)                                                                                                                 | false                     |
+| BACKEND_LOGGER_LEVEL | Logger level in backend (trace/debug/info/warn/error/fatal) (Default: debug)                                                                         | false                     |
```
Comment (shuse2) [inline packages/claim-backend/README.md]: let's set the default to be `info` so on production, we can use without changing

File: packages/claim-backend/README.md [hunk 2]
```diff
@@ -152,3 +153,19 @@ curl --location 'http://127.0.0.1:3000/rpc' \
...
 ```
+
+## Logging
+
+Claim Backend uses [log4js](https://www.npmjs.com/package/log4js) to record API call logs. 3 categories of logs will be used:
+
+- default
+  - `INFO`: General logging during the start of backend.
+- RPC
+  - `DEBUG`: When sending success RPC response to client.
```
Comment (shuse2) [inline packages/claim-backend/README.md]: I think this can be `info` level

**Review by shuse2 — CHANGES_REQUESTED** [review]

**Review by has5aan — APPROVED** [review]
LGTM, one inquiry, tried the logger.ts file in a simple setup, and discovered that the error stack is being returned as response, perhaps not an issue for our use-case as the repository is already public or how the endpoint is consumed?

**Comment by Phanco** [comment]
> LGTM, one inquiry, tried the logger.ts file in a simple setup, and discovered that the error stack is being returned as response, perhaps not an issue for our use-case as the repository is already public or how the endpoint is consumed?

Error stack will be hidden when `NODE_ENV=production` and this flag has been used in prod. I think we are good here :D

**Review by shuse2 — APPROVED** [review]

**Pull request merged**